clients/src/test/java/org/apache/kafka/clients/consumer/internals/FetchRequestManagerTest.java

```diff
@@ -0,0 +1,3630 @@
+        for (int i = 1; i <= numExceptionsExpected; i++) {
+            try {
+                recordsByPartition = fetchedRecords();
+                for (List<ConsumerRecord<byte[], byte[]>> records : recordsByPartition.values())
+                    fetchedRecords.addAll(records);
+            } catch (KafkaException e) {
+                kafkaExceptions.add(e);
+            }
+        }
+        // Should have received as much as numExceptionsExpected Kafka exceptions for tp3.
+        assertEquals(numExceptionsExpected, kafkaExceptions.size());
+    }
+
+    @Test
+    public void testSeekBeforeException() {
+        buildFetcher(OffsetResetStrategy.NONE, new ByteArrayDeserializer(),
+                new ByteArrayDeserializer(), 2, IsolationLevel.READ_UNCOMMITTED);
+
+        assignFromUser(mkSet(tp0));
+        subscriptions.seek(tp0, 1);
+        assertEquals(1, sendFetches());
+        Map<TopicIdPartition, FetchResponseData.PartitionData> partitions = new HashMap<>();
+        partitions.put(tidp0, new FetchResponseData.PartitionData()
+                        .setPartitionIndex(tp0.partition())
+                        .setHighWatermark(100)
+                        .setRecords(records));
+        client.prepareResponse(fullFetchResponse(tidp0, this.records, Errors.NONE, 100L, 0));
+        networkClientDelegate.poll(time.timer(0));
+
+        assertEquals(2, fetchedRecords().get(tp0).size());
+
+        subscriptions.assignFromUser(mkSet(tp0, tp1));
+        subscriptions.seekUnvalidated(tp1, new SubscriptionState.FetchPosition(1, Optional.empty(), metadata.currentLeader(tp1)));
+
+        assertEquals(1, sendFetches());
+        partitions = new HashMap<>();
+        partitions.put(tidp1, new FetchResponseData.PartitionData()
+                        .setPartitionIndex(tp1.partition())
+                        .setErrorCode(Errors.OFFSET_OUT_OF_RANGE.code())
+                        .setHighWatermark(100));
+        client.prepareResponse(FetchResponse.of(Errors.NONE, 0, INVALID_SESSION_ID, new LinkedHashMap<>(partitions)));
+        networkClientDelegate.poll(time.timer(0));
+        assertEquals(1, fetchedRecords().get(tp0).size());
+
+        subscriptions.seek(tp1, 10);
+        // Should not throw OffsetOutOfRangeException after the seek
+        assertEmptyFetch("Should not return records or advance position after seeking to end of topic partitions");
+    }
+
+    @Test
+    public void testFetchDisconnected() {
+        buildFetcher();
+
+        assignFromUser(singleton(tp0));
+        subscriptions.seek(tp0, 0);
+
+        assertEquals(1, sendFetches());
+        client.prepareResponse(fullFetchResponse(tidp0, this.records, Errors.NONE, 100L, 0), true);
+        networkClientDelegate.poll(time.timer(0));
+        assertEmptyFetch("Should not return records or advance position on disconnect");
+
+        // disconnects should have no affect on subscription state
```

Comment:
typo 'effect'

Reply:
De-typo-ed.

This file is largely copy and paste from the `FetcherTest` which has the same typo.